src/index.tsx

```diff
@@ -858,10 +860,13 @@ export function generateTrigger(
       let portal: React.ReactElement;
       // prevent unmounting after it's rendered
       if (popupVisible || this.popupRef.current || forceRender) {
+        if (!this.portalContainer) {
+          this.portalContainer = this.getContainer();
```

Comment:
Will this also execute multiple time as FC first render?

Reply:
`render` will be double-invoked, but `if (!this.portalContainer)` can ensure `getContainer` is called only once.

Reply:
`render` will be double-invoked, but `if (!this.portalContainer)` can ensure `getContainer` is called only once.

Reply:
So class component will be the same instance but FC will not. Right?

Reply:
So you are going to rewrite `Portal` to a class component? 😂

Reply:
Updated

Reply:
I mean comment this line:  https://github.com/react-component/trigger/pull/292/files#diff-0b5adbfe7b36e4ae2f479291e20152e33e940f7f265162d77f40f6bdb5da7405R865

For example:

> In React.StrictMode component will call render multiple time in first mount. When you want to refactor with FC, `useRef` will also init multiple time and point to different `useRef` instance which will create multiple element (This multiple render will not trigger effect so you can not clean up this in effect). But this is safe with class component since it always point to same class instance.

Reply:
Okay